vendor/github.com/samuel/go-zookeeper/zk/dnshostprovider.go

```diff
@@ -0,0 +1,88 @@
+package zk
+
+import (
+	"fmt"
+	"net"
+	"sync"
+)
+
+// DNSHostProvider is the default HostProvider. It currently matches
+// the Java StaticHostProvider, resolving hosts from DNS once during
+// the call to Init.  It could be easily extended to re-query DNS
+// periodically or if there is trouble connecting.
+type DNSHostProvider struct {
```

Comment:
this is now the default, and is a significant change from what's in place already. the old code would resolve DNS names for ZK servers upon each connect. now, by default, they're resolved ONCE per invocation of `Connect`. This is a **very** significant change that will impact people who aren't using virtual IPs for their ZK servers.